content/BCBestPractices/Readability/variables-declarations-order/index.md

```diff
@@ -0,0 +1,29 @@
++++
+title = "Variables Declarations Order"
+weight = 1430
++++
+
+<_Created by Microsoft, Described by waldo_\>
+
+## Description
+Variables declarations should be ordered by type. In general, object and complex variable types are listed first followed by simple variables. The order should be **the same as the object list in the object designer for AL objects**. Afterwards come the complex variables like RecordRef, .NET, FieldRef etc. At the end come all the simple data types in no particular order. 
+
```

Comment:
There is no 'object designer for AL objects', at least not provided by Microsoft. Isn't this referring back to the order of the object list in C/SIDE?

Maybe it's better to explicitly describe the correct list, according to CodeCop rule AA0021
https://docs.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/analyzers/codecop-aa0021

* Record
* Report
* Codeunit
* XmlPort
* Page
* Query
* Notification
* BigText
* DateFormula
* RecordId
* RecordRef
* FieldRef
* FilterPageBuilder

Rest of variables are not sorted.

Reply:
Good catch :-).